### PR TITLE
refactor(windows): clean up logging

### PR DESCRIPTION
**Pull request by mcdurdin**
* Remove unused parameters from `SendDebugMessage*` functions
* Add `SendDebugEntry` and `SendDebugExit` functions for tracking function entry/exit
* Add indenting and function names to log entries
* Remove unused debug functions
* Eliminate now-unused `hwnd` parameter in initialization functions
* Replace `Log`, `LogEntry`, `LogExit` functions with `SendDebug*` equivalents in kmtip
* Added 3 new exports: `Keyman_WriteDebugEvent2W`, `Keyman_SendDebugEntry`, `Keyman_SendDebugExit`

Many functions now have `SendDebugEntry`/`SendDebugExit` (or `return_SendDebugExit`) pairs. It is important to `SendDebugExit` on all returns from a function to keep the log indent depth consistent. In some cases I chose not to add these logging calls, e.g. on frequently called functions such as the message hooks.

# User Testing

* **TEST_WINDOWS:** Please run a basic test of Windows functionality, verifying that Keyman operates as expected.

### CODE REVIEW

**Comment by dinakaranr** [comment]
# Test Results

* **TEST_WINDOWS (passed):**  I tested this issue with the attached "keyman-18.0.67-alpha-test-11921" build on the Windows 10 & 11 OS environment: Here is my observation.
1. Installed the "keyman-18.0.67.exe" file. 
2. Keyman keyboard added in the system tray.
3. Open the keyman "Configuration" window.
4. Installed and uninstalled keyboards on the "Keyboard Layouts" tab.
5. Hotkeys works which is mentioned on the "Hotkeys" tab.
6. Keyboard switched between languages by (Left-ALT + Shift)
7. The keyboard works well in Notepad and Libreoffice. 
8. Executed the suite_baseline, Suite_capslock and suite_TSF_app.
The feature works well in Windows 10 & 11 OS environments. Thank you.

File: common/windows/cpp/src/registry.cpp [hunk 1]
```diff
@@ -206,7 +206,7 @@ BOOL RegistryReadOnly::WrapError(DWORD res)
 {
 	/*if(res != ERROR_SUCCESS)
 	{
-		SendDebugMessageFormat(0,KDS_PROGRAM,0,"RegistryFullAccess::WrapError = %d", res);
+		__logerror(0,KDS_PROGRAM,0,"RegistryFullAccess::WrapError = %d", res);
```
Comment (mcdurdin) [inline common/windows/cpp/src/registry.cpp:209]: this is commented out; just removing ref to SendDebugMessageFormat as this is a shared file

File: windows/src/engine/keyman32/K32_load.cpp [hunk 2]
```diff
@@ -60,60 +60,67 @@ BOOL GetKeyboardFileName(LPSTR kbname, LPSTR buf, int nbuf)
...
+  if (!GetKeyboardFileName(_td->lpKeyboards[i].Name, buf, 255)) {
+    return_SendDebugExit(FALSE);
+  }
+
   PWCHAR keyboardPath = strtowstr(buf);
   km_core_status err_status = km_core_keyboard_load(keyboardPath, &_td->lpKeyboards[i].lpCoreKeyboard);
   if (err_status != KM_CORE_STATUS_OK) {
-    SendDebugMessageFormat(0, sdmLoad, 0, "LoadlpKeyboard: km_core_keyboard_load failed for %ls with error status [%d]", keyboardPath, err_status);
+    SendDebugMessageFormat("km_core_keyboard_load failed for %ls with error status [%d]", keyboardPath, err_status);
     delete keyboardPath;
```
Comment (rc-swag) [inline windows/src/engine/keyman32/K32_load.cpp:90]:
```suggestion
    delete[] keyboardPath;
```

**Comment by mcdurdin** [comment]
Noteː there may be something going on with performance once debugging is enabled -- I was finding significant lag while the debugging was enabled. Needs further verification before we merge

**Comment by mcdurdin** [comment]
Need to disable `    CKMTipTextService::_PreserveAltKeys: Preserved key 100: K_oE2 0, 0` etc

File: windows/src/engine/keyman32/k32_dbg.cpp [hunk 3]
```diff
@@ -79,6 +79,7 @@ void InitDebugging() {
 
 	InitDebuggingEx(_td);
 	_td->debug_DebugInit = TRUE;
+  _td->debug_Depth = 0;
```
Comment (rc-swag) [inline windows/src/engine/keyman32/k32_dbg.cpp:82]:
```suggestion
    _td->debug_Depth = 0;
```

**Review by rc-swag — APPROVED** [review]
Just a comment about the va_end(args);
lgtm

File: windows/src/engine/kmtip/keys.cpp [hunk 4]
```diff
@@ -37,6 +35,8 @@
 #include "kmtip.h"
 #include "../../../../common/windows/cpp/include/vkeys.h"
 
+#define _DEBUG_LOGKEY
```
Comment (mcdurdin) [inline windows/src/engine/kmtip/keys.cpp:38]:
```suggestion
//#define _DEBUG_LOGKEY
```

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.77-alpha](https://keyman.com/downloads/releases/alpha/18.0.77)

**Pull request merged**